plutus-tx-plugin/test/BuiltinList/Budget/Spec.hs

```diff
@@ -19,31 +19,254 @@ tests :: TestNested
+uniqueElementNothing = $$(compile [|| \_ -> L.uniqueElement L.empty ||])
+
+revAppend :: CompiledCode (L.BuiltinList Integer -> L.BuiltinList Integer)
+revAppend = $$(compile [|| \xs -> L.revAppend xs xs ||])
+
+reverse :: CompiledCode (L.BuiltinList Integer -> L.BuiltinList Integer)
+reverse = $$(compile [|| \xs -> L.reverse xs ||])
+
+replicate :: CompiledCode (L.BuiltinList Integer -> L.BuiltinList Integer)
+replicate = $$(compile [|| \_ -> L.replicate 10 0 ||])
+
+findIndexJust :: CompiledCode (L.BuiltinList Integer -> Maybe Integer)
+findIndexJust = $$(compile [|| \xs -> L.findIndex (P.== 4) xs ||])
+
+findIndexNothing :: CompiledCode (L.BuiltinList Integer -> Maybe Integer)
+findIndexNothing = $$(compile [|| \xs -> L.findIndex (P.== 99) xs ||])
+
+zipWith :: CompiledCode (L.BuiltinList Integer -> L.BuiltinList Integer)
+zipWith = $$(compile [|| \xs -> L.zipWith (P.+) xs xs ||])
+
+headOk :: CompiledCode (L.BuiltinList Integer -> Integer)
+headOk = $$(compile [|| \xs -> L.head xs ||])
+
+headEmpty :: CompiledCode (L.BuiltinList Integer -> Integer)
+headEmpty = $$(compile [|| \_ -> L.head L.empty ||])
+
+lastOk :: CompiledCode (L.BuiltinList Integer -> Integer)
+lastOk = $$(compile [|| \xs -> L.last xs ||])
+
+lastEmpty :: CompiledCode (L.BuiltinList Integer -> Integer)
+lastEmpty = $$(compile [|| \_ -> L.last L.empty ||])
+
+tailOk :: CompiledCode (L.BuiltinList Integer -> L.BuiltinList Integer)
+tailOk = $$(compile [|| \xs -> L.tail xs ||])
+
+tailEmpty :: CompiledCode (L.BuiltinList Integer -> L.BuiltinList Integer)
+tailEmpty = $$(compile [|| \_ -> L.tail L.empty ||])
+
+take :: CompiledCode (L.BuiltinList Integer -> L.BuiltinList Integer)
+take = $$(compile [|| \xs -> L.take 5 xs ||])
+
+drop :: CompiledCode (L.BuiltinList Integer -> L.BuiltinList Integer)
+drop = $$(compile [|| \xs -> L.drop 5 xs ||])
+
+dropWhile :: CompiledCode (L.BuiltinList Integer -> L.BuiltinList Integer)
+dropWhile = $$(compile [|| \xs -> L.dropWhile (P.< 5) xs ||])
+
+elemBy :: CompiledCode (L.BuiltinList Integer -> Bool)
+elemBy = $$(compile [|| \xs -> L.elemBy (P.<=) 0 xs ||])
+
+nub :: CompiledCode (L.BuiltinList Integer -> L.BuiltinList Integer)
+nub = $$(compile [|| \xs -> L.nub (L.append xs xs) ||])
+
+nubBy :: CompiledCode (L.BuiltinList Integer -> L.BuiltinList Integer)
+nubBy = $$(compile [|| \xs -> L.nubBy (P.<=) xs ||])
+
+l1 :: CompiledCode (L.BuiltinList Integer)
+l1 = liftCodeDef $ toBuiltin ([1 .. 10] :: [Integer])
+
+l2 :: CompiledCode (L.BuiltinList P.BuiltinBool)
+l2 = liftCodeDef $ toBuiltin ([True, False, True, False] :: [Bool])
+
+l3 :: CompiledCode (L.BuiltinList (P.BuiltinPair Integer Integer))
+l3 = liftCodeDef $ toBuiltin ([ (1, 2), (3, 4), (5, 6) ] :: [(Integer, Integer)])
+
+l4 :: CompiledCode (L.BuiltinList (L.BuiltinList Integer))
+l4 = liftCodeDef $ toBuiltin ([[1, 2], [3, 4]] :: [[Integer]])
+
+-- TODO The following functions cannot compile because they require implementation of
```

Comment:
You can keep this code:
- in a patch
- in a stash
- in a branch

Please don't keep it in production code.